src/main/java/ch/securify/CompilationHelpers.java

```diff
@@ -83,18 +88,39 @@ private static String readFile(String path) throws IOException {
         return new String(encoded, Charset.defaultCharset());
     }
 
-    static JsonObject compileContracts(String filesol) throws IOException, InterruptedException {
+    static JsonObject compileContracts(String filesol) throws IOException, InterruptedException, RuntimeException {
         ProcessBuilder p = new ProcessBuilder("solc", "--combined-json", "abi,ast,bin-runtime,srcmap-runtime", filesol);
 
         File f = File.createTempFile("securify_compilation_", ".json");
         f.deleteOnExit();
 
         final Process process = p.redirectOutput(f).start();
 
+        Thread t = new Thread(){
+            public void run(){
+                try {
+                    InputStreamReader isr = new InputStreamReader(process.getErrorStream());
+                    BufferedReader br = new BufferedReader(isr);
+                    String line = br.readLine();
+                    if (!line.contains("Error")) {
+                        return;
```

Comment:
This will only output anything if the first line of the error stream contains the word "Error", correct? Why this limitation?

Reply:
> Proposed solution for #24
> 
> Securify reports a generic IOException when solc fails.
> 
> This PR addresses that by collecting the solc Process' ErrorStream and piping it out to System.err
> 
> A check to see that the first line has 'Error' is so that the Warnings from solc are not printed when there are no errors.

The last sentence :) when there were no errors then when you run the command it would print out the warnings from solc as well as the json output which is easily averted by just spliting the error stream off, I just had it like that coz i was piping all the output into a json file

Reply:
I was originally checking the whole `stacktrace` string for 'Error' but that seemed redundant since solc always reports the error first

Reply:
I think it is OK to print warnings as long as they go to stderr.